Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationManager.kt

```diff
@@ -0,0 +1,77 @@
+package de.rki.coronawarnapp.ccl.dccwalletinfo.calculation
+
+import de.rki.coronawarnapp.ccl.configuration.storage.CCLConfigurationRepository
+import de.rki.coronawarnapp.ccl.dccwalletinfo.storage.DccWalletInfoRepository
+import de.rki.coronawarnapp.covidcertificate.booster.BoosterRulesRepository
+import de.rki.coronawarnapp.covidcertificate.common.certificate.CertificatePersonIdentifier
+import de.rki.coronawarnapp.covidcertificate.person.core.PersonCertificates
+import de.rki.coronawarnapp.covidcertificate.person.core.PersonCertificatesProvider
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import timber.log.Timber
+import javax.inject.Inject
+
+class DccWalletInfoCalculationManager @Inject constructor(
+    private val cclConfigurationRepository: CCLConfigurationRepository,
+    private val boosterRulesRepository: BoosterRulesRepository,
+    private val personCertificatesProvider: PersonCertificatesProvider,
+    private val dccWalletInfoRepository: DccWalletInfoRepository,
+    private val calculation: DccWalletInfoCalculation,
+    private val timeStamper: TimeStamper,
+    @AppScope private val appScope: CoroutineScope,
+) {
+
+    fun setup() {
+        Timber.d("setup()")
+        boosterRulesRepository.rules.onEach {
```

Comment:
I think this is not necessary. If new booster rules are downloaded, `triggerCalculation()` will anyway be called from the CCLConfigUpdaterScheduler.